src/foam/classloader/WebModelFileDAO.js

```diff
@@ -52,10 +52,19 @@ foam.CLASS({
         var model;
         var foamCLASS = foam.CLASS;
 
-        foam.CLASS = function(m) {
-          var cls = m.class ? foam.lookup(m.class) : foam.core.Model;
-          model = cls.create(m, self);
-          foam.CLASS = foamCLASS;
+        foam.CLASS = function(clsDef) {
+          var classOfModel = clsDef.class ? foam.lookup(clsDef.class) :
```

Comment:
Oops. New changes were added to #259. PTAL there. I'll make the code here match.